### PR TITLE
Migrate to go mod

### PR DESCRIPTION
**Pull request by ruromero**

### CODE REVIEW

**Comment by ruromero** [comment]
Revision of https://github.com/RHsyseng/operator-utils/pull/36

**Comment by jeremyary** [comment]
@tchughesiv you mind throwing in a review here? I've not dealt with the go mod conversion enough to see any possible downfalls, but I'm not seeing any issues otherwise

**Review by ricardozanini — APPROVED** [review]
I know I haven't been invited to the party, but I've done this before in the Kogito Operator. Looks good to me. :)

File: .travis.yml [hunk 1]
```diff
@@ -2,14 +2,8 @@ language: go
...
   - 1.12.x
 
 env:
-  - DEP_RELEASE_TAG="v0.5.1"
-
-before_install:
-  # Setup dependency management tool
-  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
-  - dep version
+  - GO111MODULE=on
```
Comment (tchughesiv) [inline .travis.yml:9]: add this to ensure vendor dir used for tests, vet, builds, etc...
`- GOFLAGS=-mod=vendor`

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-dep:
-	dep ensure -v
+.PHONY: mod
+mod:
+	go mod tidy && go mod vendor
 
 .PHONY: format
-format: dep
+format: mod
 	go fmt ./...
```
Comment (tchughesiv) [inline Makefile:12]: add `-mod=vendor` flag to `vet` and `test` commands in Makefile

Comment (tchughesiv) [reply Makefile]: add to vet and test as well...

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (ruromero) [reply Makefile:12]: Thanks @tchughesiv 
Updated

Comment (ruromero) [reply Makefile:12]: If it is added to the Makefile why add it to the travis file? We will be setting it twice which is not allowed.
I have added it only to the Makefile

**Comment by ruromero** [comment]
@jeremyary @tchughesiv can we get this merged?
Thanks

File: Makefile (hunk 2 above)
Comment (tchughesiv) [reply Makefile:12]: Sounds good for travis file... likely need to add to build command though.

**Review by tchughesiv — APPROVED** [review]

**Pull request merged**